src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -0,0 +1,472 @@
+                }
+            }
+            l.log("Connection interrupted");
+
+            cancelUpdateSubscriptionsTask();
+        }
+
+        @Override
+        public void onStopped(Mqtt5Client client, OnStoppedReturn onStoppedReturn) {
+            client.close();
+        }
+    };
+
+    @Getter(AccessLevel.PACKAGE) // for testing
+    private final Mqtt5ClientOptions.PublishEvents publishEventsCallback = (client, publishReturn) ->
+            this.messageHandler.accept(MqttMessage.fromSpoolerV5Model(
+                    Publish.fromCrtPublishPacket(publishReturn.getPublishPacket())));
+
+    /**
+     * Construct a LocalMqtt5Client.
+     *
+     * @param brokerUri          broker uri
+     * @param clientId           client id
+     * @param mqttClientKeyStore KeyStore for MQTT Client
+     * @param executorService    Executor service
+     * @throws MessageClientException if unable to create client for the mqtt broker
+     */
+    public LocalMqtt5Client(@NonNull URI brokerUri,
+                            @NonNull String clientId,
+                            MQTTClientKeyStore mqttClientKeyStore,
+                            ExecutorService executorService) throws MessageClientException {
+        this.brokerUri = brokerUri;
+        this.clientId = clientId;
+        this.executorService = executorService;
+
+        boolean isSSL = "ssl".equalsIgnoreCase(brokerUri.getScheme());
+
+        long port = brokerUri.getPort();
+        if (port < 0) {
+            port = isSSL ? DEFAULT_SSL_MQTT_PORT : DEFAULT_TCP_MQTT_PORT;
+        }
+
+        Mqtt5ClientOptions.Mqtt5ClientOptionsBuilder builder =
+                new Mqtt5ClientOptions.Mqtt5ClientOptionsBuilder(brokerUri.getHost(), port)
+                        .withLifecycleEvents(connectionEventCallback)
+                        .withPublishEvents(publishEventsCallback)
+                        .withSessionBehavior(Mqtt5ClientOptions.ClientSessionBehavior.REJOIN_POST_SUCCESS)
+                        .withOfflineQueueBehavior(Mqtt5ClientOptions.ClientOfflineQueueBehavior.FAIL_ALL_ON_DISCONNECT)
+                        .withConnectOptions(new ConnectPacket.ConnectPacketBuilder()
+                                .withRequestProblemInformation(true)
+                                .withClientId(clientId).build())
+                        // TODO configurable?
+                        .withMaxReconnectDelayMs((long)MAX_WAIT_RETRY_IN_SECONDS * 1000)
+                        .withMinReconnectDelayMs((long)MIN_WAIT_RETRY_IN_SECONDS * 1000);
+
+        if (isSSL) {
+            mqttClientKeyStore.listenToCAUpdates(this::reset);
+            this.tlsContextOptions = TlsContextOptions.createWithMtlsJavaKeystore(mqttClientKeyStore.getKeyStore(),
+                    KEY_ALIAS, Arrays.toString(DEFAULT_KEYSTORE_PASSWORD));
+            this.tlsContext = new TlsContext(tlsContextOptions);
+            builder.withTlsContext(tlsContext);
+        }
```

Comment:
none of this can be in the constructor, right? If the customer changes from tls to non-tls we won't create a new LocalMqtt5Client, we'd just stop it and start it again I believe.

Reply:
new LocalMqtt5Client is created on startup, so should be good